### ISSUE TITLE
Mapbox studio won't open in Ubuntu 14.04 LTS

### ISSUE THREAD

**Issue by brianbancroft** [mapbox/mapbox-studio-classic#1465]
Hi there, 

I've had problems installing and running Mapbox Studio Classic on a fresh install of Ubuntu 14.04 LTS. When I download the program and run Atom, the program opens for a half-second then closes.

I am unable to find the app.log or shell.log in the mapbox studio directory or after a general search, but I've found a npm.log in one of the subdirectories. 

```
0 info it worked if it ends with ok
1 verbose cli [ 'node', '/usr/local/bin/npm', 'start' ]
2 info using npm@1.4.23
3 info using node@v0.10.31
4 error Error: ENOENT, open '/Users/duncangraham/Documents/Mapbox/mapbox-studio/templates/package.json'
5 error If you need help, you may report this *entire* log,
5 error including the npm and node versions, at:
5 error     <http://github.com/npm/npm/issues>
6 error System Darwin 14.1.0
7 error command "node" "/usr/local/bin/npm" "start"
8 error cwd /Users/duncangraham/Documents/Mapbox/mapbox-studio/templates
9 error node -v v0.10.31
10 error npm -v 1.4.23
11 error path /Users/duncangraham/Documents/Mapbox/mapbox-studio/templates/package.json
12 error code ENOENT
13 error errno 34
14 verbose exit [ 34, true ]
```

Thanks for any assistance!

**Comment by wilhelmberg** [comment]
If you cloned from Github, you probably forgot to run `npm install`:
https://github.com/mapbox/mapbox-studio-classic#install-from-source

If you downloaded from
https://www.mapbox.com/mapbox-studio-classic/#linux
extract the archive, `cd` into the extracted directory and run `./atom`.

**Comment by brianbancroft** [comment]
I attempted both methods before writing this issue. 

I've since reformatted and installed a fresh version of studio-classic on a fresh install of Ubuntu. The problem persists: I attempt to run Mapbox Studio via ./mapbox-studio-linux-x64-v0.3.0/atom, the window pops up for a fraction of a second, then closes again.

**Comment by wilhelmberg** [comment]
@brianbancroft could you check again in `~/.mapbox-studio`, if you find `app.log` and/or `shell.log` and paste their contents here.

Please try the following to see, if we get more information this way:

```
cd ./mapbox-studio-linux-x64-v0.3.0/resources/app
./vendor/node index.js
```

Is there an error message or a message like `Mapbox Studio @ http://localhost:3000/`?
If error message, post it here.
If no error message, open your browser (preferably Chrome) at http://localhost:3000

**Comment by brianbancroft** [comment]
Hi there, 

After running "node index.js" in the directory you've passed my way, I find the following problem:

```
The program 'node' can be found in the following packages:
 * node
 * nodejs-legacy
Try: sudo apt-get install <selected package>
```

In the ./mapbox-studio-linux-x64-v.0.3.0/ directory, I've found no app.log or shell.log file. I believe I may be looking in the wrong directory, but I'm unsure where else to look.

**Comment by wilhelmberg** [comment]
@brianbancroft you forgot `./vendor/`:

<pre>
<b>./vendor</b>/node index.js
</pre>

**Comment by wilhelmberg** [comment]
The log files are not in `./mapbox-studio-linux-x64-v.0.3.0/`, they should be in

<pre>
<b>~/</b>.mapbox-studio
</pre>

**Comment by brianbancroft** [comment]
Thanks for sticking with me so far :)

So for the first reply I didn't forget the vendor directory. 

For the second reply, thanks for staying with me. I've found what you're looking for. App.log only has one line: `<Sun, 20 Sep 2015 13:34:09 GMT> Mapbox Studio @ http://localhost:3000/`. I tried running this on Firefox while I was waiting for chrome to download. Here, I was able to log in, and now the program works as desired. 

Thanks for walking me through!
